### PR TITLE
Eq instance for Graph

### PR DESCRIPTION
**Pull request by gelisam**

### CODE REVIEW

File: test/Data/Graph/WrapperSpec.hs [hunk 1]
```diff
@@ -21,3 +21,29 @@ spec = do
       it "returns the empty list" $ do
         reachableVertices (fromList [] :: Graph Integer ()) 0
           `shouldBe` []
+
+  describe "Eq Graph" $ do
+    context "when the graphs are identical" $ do
+      it "returns True" $ do
+        let g1 = fromList [('a','a',['a','b','c']), ('b','b',['b','c']), ('c','c',['a'])]
```
Comment (soenkehahn) [inline test/Data/Graph/WrapperSpec.hs]: i think the whitespace here is a inconsistent. Sometimes a comma is followed by a space, sometimes it's not. Could you make sure every comma is followed by a space, please?

File: test/Data/Graph/WrapperSpec.hs [hunk 2]
```diff
@@ -21,3 +21,29 @@ spec = do
...
+        let g2 = fromList [('a','a',['a','b','c']), ('b','b',['b','c']), ('c','c',['a'])]
+        (g1 == g2)
+          `shouldBe` True
```
Comment (soenkehahn) [inline test/Data/Graph/WrapperSpec.hs]: I would put the `shouldBe` expression in one line. Also, I think, you don't need the parenthesis around `g1 == g2`, although I'm fine with keeping them, if you prefer.

**Comment by gelisam** [comment]
> Some time ago I thought about adding an `Eq` instance. Back then I didn't, because I thought it was confusing in cases like this:
> 
>     fromList [(1,'a',[2]), (2,'b',[])] == fromList [(2,'a',[1]), (1,'b',[])]
> 
> So the graphs are in a sense the same (two nodes `a` and `b` and one edge `(a, b)`). But the indices differ so that your `Eq` instance yields `False`. What do you think about this?

I did think about this, and my conclusion is that _if_ your `Graph` type was only indexed by `v`, this would be a legitimate concern, but since they're indexed by `i` as well, I would not expect two nodes with different `i`s to be considered equal.

File: src/Data/Graph/Wrapper/Internal.hs [hunk 3]
```diff
@@ -33,6 +34,15 @@ data Graph i v = G {
...
 
+-- The arrays are all sorted by 'i'. The only part whose order could differ between identical graphs
+-- is the order in which 'graph' lists its edges, so we should sort those.
+instance (Ord i, Eq v) => Eq (Graph i v) where
+  g1 == g2 = fmap sort (graph g1) == fmap sort (graph g2)
+          && indexGVertexArray g1 == indexGVertexArray g2
+          && gVertexVertexArray g1 == gVertexVertexArray g2
+
+
+
```
[the remark this answers is not in the document]
Comment (gelisam) [reply src/Data/Graph/Wrapper/Internal.hs]: Oops! Now using a single line.

File: test/Data/Graph/WrapperSpec.hs [hunk 4]
```diff
@@ -21,3 +21,29 @@ spec = do
...
+    context "when the vertices are listed in a different order" $ do
```
[the remark this answers is not in the document]
Comment (gelisam) [reply test/Data/Graph/WrapperSpec.hs]: added.

File: test/Data/Graph/WrapperSpec.hs (hunk 1 above)
Comment (gelisam) [reply test/Data/Graph/WrapperSpec.hs]: added.

File: test/Data/Graph/WrapperSpec.hs (hunk 2 above)
Comment (gelisam) [reply test/Data/Graph/WrapperSpec.hs]: fixed.

File: test/Data/Graph/WrapperSpec.hs [hunk 5]
```diff
@@ -21,3 +21,34 @@ spec = do
...
+    context "when the graphs are identical" $ do
+      it "returns True" $ do
+        let g1 = fromList [('a', 'a', ['a', 'b', 'c']), ('b', 'b', ['b', 'c']), ('c', 'c', ['a'])]
+        let g2 = fromList [('a', 'a', ['a', 'b', 'c']), ('b', 'b', ['b', 'c']), ('c', 'c', ['a'])]
+        g1 == g2 `shouldBe` True
+
+    context "when the graphs are isomorphic" $ do
+      it "returns False" $ do
+        let g1 = fromList [('1','a',['2']), ('2','b',[])]
+        let g2 = fromList [('2','a',['1']), ('1','b',[])]
```
Comment (soenkehahn) [inline test/Data/Graph/WrapperSpec.hs]: Sorry, could you add in the spaces after the commas again? :)

Comment (gelisam) [reply test/Data/Graph/WrapperSpec.hs]: oops! old habits die hard :)

fixed.

**Comment by soenkehahn** [comment]
Merged. Thanks again! I created a PR for a new hackage release: #7. Could you take a look, please?

**Pull request merged**